datavec/datavec-python/src/main/java/org/datavec/python/PythonTransform.java

```diff
@@ -0,0 +1,171 @@
+package org.datavec.python;
+
+import org.json.simple.JSONArray;
+import org.json.simple.JSONObject;
+import org.json.simple.parser.JSONParser;
+import org.json.simple.parser.ParseException;
+
+import javax.annotation.Nullable;
+
+import java.io.BufferedReader;
+import java.io.FileReader;
+import java.io.FileWriter;
+import java.io.IOException;
+import java.util.List;
+import java.util.concurrent.ExecutionException;
+
+public class PythonTransform {
+    private String setupCode;
+    private String execCode;
+    private String code;
+    private String name;
+    private PythonVariables pyInputs;
+    private PythonVariables pyOutputs;
+
+    private void parseSetupAndExecCode() throws Exception{
```

Comment:
this is a confusing name (although I see how you got it). It suggests that you parse the setup and exec the code.

A doc string might help. Maybe find an unambiguous convention.

Also, what's the setup tag for? Can we have a test for it?

Reply:
The "setup" part of the code is whatever you need to do only once (imports, loading stuff to memory). We do it once and persist that state in memory so that we dont have to do it per transaction. I havent committed the tests because of the issue i discussed in our call (regarding arrays getting GCed too early sometimes). Will add tests once it works.